lib/vector-core/src/event/vrl_target.rs

```diff
@@ -106,12 +108,41 @@ impl vrl_core::Target for VrlTarget {
                             metric.data.kind = MetricKind::try_from(value)?;
                             return Ok(());
                         }
+                        ["counter", "value"] => {
+                            let new_value = match value {
+                                vrl_core::Value::Integer(i) => Ok(i as f64),
+                                vrl_core::Value::Float(f) => Ok(*f),
+                                _ => Err(format!(
+                                    "expected integer or float, found: {}",
+                                    value.kind()
+                                )),
+                            }?;
+
+                            if let MetricValue::Counter { ref mut value } = metric.data.value {
```

Comment:
This should probably error if it isn't a counter, although it's probably fairly moot since VRL ignores the errors here..

Reply:
Or should it allow you to turn a guage into a counter?

Reply:
Yeah, true, we could return an error here.

I think we should avoid allowing metric type conversion for now and just allow reading / writing values to reduce scope.